Sources/SwiftProtobuf/NameMap.swift

```diff
@@ -8,147 +8,217 @@
 //
 // -----------------------------------------------------------------------------
 
-// Proto names are always ASCII and are always literals, so we can improve the
-// performance of some of our parsing by storing them as pointers to the
-// underlying bytes of a StaticString to avoid the overhead of creating and
-// comparing Unicode Strings.
+import Darwin
 
-// Constants for FNV hash http://tools.ietf.org/html/draft-eastlake-fnv-03
-private let i_2166136261 = Int(bitPattern: 2166136261)
-private let i_16777619 = Int(16777619)
-
-/// A simple byte wrapper than can hold (and hash/compare) either a
-/// pointer/count to bytes in memory. We don't worry about retention because the
-/// data is assumed to either (1) come from literal strings in the data segment
-/// or (2) not escape the lifetime of the `String` object from which the data
-/// was taken. This is inherently unsafe; use caution when using this type.
-private struct ASCIIName: Hashable {
+/// This must be exactly the same as the corresponding code in the
+/// protoc-gen-swift code generator.  Changing it will break
+/// compatibility of the library with older generated code.
```

Comment:
I'm ambivalent about that.

I see two arguments _against_ sharing a single implementation of `toJsonFieldName`:

1. It's behavior is part of the contract between the library and the generated code; changing it will break compatibility with older library or generated code.  If having one shared instance might tempt someone to assume it's safe to change it, then it might be better to not share it.

2. I don't see a reason for this to be in the library API.

Do you think it should be shared?